core/src/main/scala/com/mesosphere/usi/core/util/RateLimiterFlow.scala

```diff
@@ -0,0 +1,70 @@
+package com.mesosphere.usi.core.util
+
+import java.time.{Clock, Duration => JavaDuration}
+
+import akka.NotUsed
+import akka.stream.scaladsl.Flow
+import akka.stream.stage._
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+
+import scala.concurrent.duration.FiniteDuration
+
+private[usi] class RateLimiterFlow[U] private (rate: FiniteDuration, clock: Clock) extends GraphStage[FlowShape[U, U]] {
```

Comment:
Nice. I actually wanted to point out that this was in Marathon already. What do you think about opening up a PR in Marathon to replace the Marathon implementation with this? We already have a test dependency on USI.

Reply:
I'd rather not right now